vms/platformvm/block/builder/builder.go

```diff
@@ -188,10 +182,10 @@ func (b *builder) setNextBuildBlockTime() {
 		return
 	}
 
-	if _, err := b.buildBlock(); err == nil {
-		// We can build a block now
+	now := b.txExecutorBackend.Clk.Time()
+	if !b.nextStakerChangeTime.After(now) {
+		// Block needs to be issued to advance time.
 		b.Mempool.RequestBuildBlock(true /*=emptyBlockPermitted*/)
```

Comment:
The PR description says: `setNextBuildBlockTime now no longer builds a block. It only is responsible for updating the nextStakerChangeTime field.` We're not calling `buildBlock` in here anymore, but we're _sort of_ building a block by calling `b.Mempool.RequestBuildBlock`.

Reply:
Should we return here? The rest of the work we do in this method seems unnecessary because `BuildBlock` is about to be invoked by the engine, and therefore the `nextStakerChangeTime` doesn't matter.

Reply:
Re-worked this function a bit and now return early